samples/security/spire/README.md

```diff
@@ -0,0 +1,72 @@
+# Integrating SPIRE as a CA through Envoy's SDS API
+
+This sample deploys a setup of [SPIRE](https://github.com/spiffe/spire) (the SPIFFE Runtime Environment) as an example of integrating with [Envoy's SDS](https://www.envoyproxy.io/docs/envoy/latest/configuration/security/secret) API. For more information
+on SPIFFE and SPIRE specs, refer to the [SPIFFE Overview](https://spiffe.io/docs/latest/spiffe-about/overview/).
+
+Once SPIRE is deployed and integrated with Istio, this sample deploys a modified version of the [sleep](/samples/sleep/README.md) service and validates that its [identity](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid) was issued by SPIRE.
+
+See [Istio CA Integration with SPIRE](https://istio.io/latest/docs/ops/integrations/spire) for further details about this integration.
+
+## Deploy the integration
+
+1. Deploy SPIRE. For proper socket injection, this **must** be done prior to installing Istio in your cluster:
+
+  ```bash
+  $ kubectl apply -f spire-quickstart.yaml
+  ```
+
+1. Ensure that the deployment is completed before moving to the next step. This can be verified by waiting on the `spire-agent` pod to become ready:
+
+  ```bash
+  $ kubectl wait pod --for=condition=ready -n spire -l app=spire-agent
+  ```
+
+1. Use the configuration profile provided to install Istio (requires istioctl v1.15 or newer):
```

Comment:
1.15?

Reply:
My apologies, I forgot I should've already left a comment about this:

I have two small concerns regarding this:

1. Since this is a new feature, is it a good idea to point out the requirement of the version? Or could that be omitted?

2. If deciding to go with the version requirement, as I didn't know at what version it would end up at, should a placeholder for the version be specified instead of a "guess" to avoid confusion?

Reply:
I think it is reasonable to require a given istio version.  If this lands in 1.14, then it should require Istio 1.14.

Reply:
Since you have this usage in a 1.14 doc PR, I am making an assumption this works in 1.14 and that the comment should say 1.14. If this won't work in 1.14, but needs 1.15, then we shouldn't have a doc PR yet, wait until after istio.io branches before merging.  I hope this makes sense. In this repo, it might be hard to tell version, while the user should be able to tell version from the docs (preliminary.istio.io is currently 1.14 docs while istio.io is 1.13 docs).

Reply:
Thanks @linsun and @ericvn. I guess I was just being very cautious regarding how long it could take for the main PR to get approved/merged. I'll update this once we have the main PR merged to ensure the correct version.

Reply:
Okay I see what you mean now @ericvn, regarding the docs PR targeting 1.14. Seems like its well stablished then, I'll update accordingly.